segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -399,6 +400,17 @@ public void createSegment(CreateSegment createStreamsSegment) {
                 });
     }
 
+    private Void handleReadException(long readOffset, String segment, Throwable u) {
```

Comment:
I think the reason why you are creating a separate `handleReadException` rather than deal with `CancellationException` in `handleException` is that you need to know that it is a `CancellationException` out of a read request and you don't want to handle generally `CancellationException` in `handleException`.

Assuming this reasoning is correct, `CancellationException` by itself is not giving enough information for us to handle the exception appropriately. The overall flow looks ok, but the idea of handling the exception separately is not aligned with the rest of the code in PRP. 

Two options I can think of to deal with this:
1- The exception that the segment store throws should have a narrower scope, e.g., `ReadCancellationException`.
2- In `handleReadResult`, we wrap `CancellationException` with something else, e.g., `ReadCancellationException` so that we can disambiguate in `handleException`.

The idea would be to catch and handle the exception in `handleException` rather than add this new method.

Reply:
ok, improved `handleException` based on the review comment.